tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/tuples/TupleAccessExprTest.java

```diff
@@ -201,6 +201,14 @@ public void testTupleAccessUsingUnionWithFiniteTypesNegative() {
         BRunUtil.invoke(compileResult, "testTupleAccessUsingUnionWithFiniteTypesNegative");
     }
 
+    @Test
+    public void testTupleAccessUsingCustomTypes() {
+        BRunUtil.invoke(compileResult, "testTupleAccessWithCustomType");
+        BRunUtil.invoke(compileResult, "testTupleAccessWithCustomType2");
+        BRunUtil.invoke(compileResult, "testTupleAccessWithCustomUnionTypes");
+        BRunUtil.invoke(compileResult, "testTupleAccessWithCustomReadonlyUnionTypes");
```

Comment:
Let's use a data provider instead. That way all 4 tests will run even if one fails.

Reply:
Updated 0b97792c352b9e0d5bb788ace062d3713de2d277